### PR TITLE
Clear encryption data when no longer needed

### PR DESCRIPTION
**Pull request by dlmarion**

### CODE REVIEW

**Comment by dlmarion** [comment]
There are still some issues with this PR, just looking to get feedback on the approach.

File: core/src/main/java/org/apache/accumulo/core/spi/crypto/AESCryptoService.java [hunk 1]
```diff
@@ -200,6 +207,15 @@ public void setEncFek(byte[] encFek) {
...
 
+    @Override
+    public void close() throws Exception {
+      cryptoServiceName = null;
+      cryptoServiceVersion = null;
+      keyManagerVersion = null;
+      kekId = null;
+      Arrays.fill(encFek, (byte) 0);
+    }
+
```
Comment (milleruntime) [inline core/src/main/java/org/apache/accumulo/core/spi/crypto/AESCryptoService.java:218]: These changes look good so far. I like using AutoCloseable, that is a good idea.

File: core/src/main/java/org/apache/accumulo/core/file/rfile/bcfile/BCFile.java [hunk 2]
```diff
@@ -635,8 +636,10 @@ public long getRawSize() {
...
       if (version.equals(API_VERSION_1)) {
         LOG.trace("Found a version 1 file to read.");
-        decryptionParams = new NoFileEncrypter().getDecryptionParameters();
         this.decrypter = new NoFileDecrypter();
+        try (FileEncrypter enc = new NoFileEncrypter()) {
+          decryptionParams = enc.getDecryptionParameters();
+        }
       } else {
         // read crypto parameters and get decrypter
         this.in.seek(offsetCryptoParameters);
```
Comment (milleruntime) [inline core/src/main/java/org/apache/accumulo/core/file/rfile/bcfile/BCFile.java:645]: We might want to put some checks in to ensure that the CryptoService is open so we don't write null values to the file.

**Comment by dlmarion** [comment]
> I think this should be held back until that one is merged.

I have no issue with that

> However, it seems quite large, and some of it probably conflicts with the design changes in https://github.com/apache/accumulo/pull/2197.

IIRC most of the changes here have to do with the fact that CryptoService is now AutoClosable

**Comment by milleruntime** [comment]
> > However, it seems quite large, and some of it probably conflicts with the design changes in #2197.
> 
> IIRC most of the changes here have to do with the fact that CryptoService is now AutoClosable

The changes in #2197 are also incomplete so that would be another reason to wait and could be why this PR looks so big.

**Comment by dlmarion** [comment]
Closing this for now as I'm not sure that adding AutoCloseable to the CryptoService is the right thing to do.

**Pull request closed without merging**